pykoi/rlhf/supervised_finetuning.py

```diff
@@ -208,15 +287,15 @@ def create_datasets(self, tokenizer, args):
         train_dataset = ConstantLengthDataset(
```

Comment:
One caveat here is that `ConstantLengthDataset` always prepare your dataset into `seq_length` by breaking one coherent c + q + a into multiple data point if len(c + q + a) > seq_length

While `DataCollatorForCompletionOnlyLM` implementation for SFTTrainer is to train mask query and train an object for response only.

However, I am a bit confused that your dataset is not prepared to train on response only (mask out query) but still casual langauge model object for next token.